framework/src/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -1731,7 +1734,11 @@ public final boolean hasOption(String name) {
     @Override
     public final String getOption(String name, String def) {
 
-        if (!this.getSupportedOptions().contains(name)) {
+        if (supportedOptions == null) {
```

Comment:
Wouldn't it be better to move this cache into `getSupportedOptions`? Then all callers of the method benefit from the cache. This method would stay unchanged.